lib/rex/post/meterpreter/extensions/stdapi/railgun/library.rb

```diff
@@ -315,6 +316,10 @@ def build_response(packet, function, layouts, arch)
         return_hash['return'] = (rec_return_value != 0)
       when 'VOID'
         return_hash['return'] = nil
+      when 'PCHAR'
+        return_hash['return'] = read_string(rec_return_value)
+      when 'PWCHAR'
+        return_hash['return'] = read_wstring(rec_return_value)
```

Comment:
I know you got these from me, but I think we need to add a check here along the lines of:

```
return_hash['return'] = rec_return_value == 0 ? nil : read_string(rec_return_value)
```

That will ensure we don't crash on functions like [`ldap_next_attribute`](https://learn.microsoft.com/en-us/windows/win32/api/winldap/nf-winldap-ldap_next_attribute) that will sometimes return NULL.

Reply:
Good point I'll add that in now

Reply:
Implementation made in e26ab2c202fe6bc6b90e383bee6e20233966c6e5